### PR TITLE
Punctuation refactor

### PR DESCRIPTION
**Pull request by kendonB**
This PR is a rewrite of the punctuation list which allows the dictionary of specs to characters to be accessible by other parts of caster.

It includes new specs for:

- backtick "`" is "ticky" and a pair is "tickris"
- double quote """ is "chocky"
- single quote "'" is "chicky" (and still apostrophe as an option)

It also adds single enclosure characters:
```
    "left prekris":                 "(",
    "right prekris":                ")",
    "left brax":                   "[",
    "right brax":                  "]",
    "left angle":                  "<",
    "right angle":                 ">",
    "left curly":                  "{",
    "right curly":                 "}",
```

It also adds "long" options for everything and repeat options for everything. These were only implemented for some before.

It also adds "bam" for ". " i.e. "bam" is equivalent to "dot ace".

It also reorders both the quick ref and source code order to be the same as ASCII so that it's easier to verify they're all there (they weren't in quick ref for example).

It also styles the quick ref so that the characters are in typewriter font.

Finally, it allows repeats of up to 100 for "ace" (others are up to 10).

- [x] Kenny test everything
- [ ] Someone else test
- [x] Get input on new specs
- [x] Add docs

### CODE REVIEW

**Comment by kendonB** [comment]
@mrob95 @LexiconCode, thoughts on these new specs?

**Comment by LexiconCode** [comment]
I think the new specs are fine. If something needs tweaking we can change it as we move along.

**Comment by kendonB** [comment]
lace / race get confused for ace too often. Any thoughts for alternatives to these? @alexboche?

**Comment by mrob95** [comment]
I'm not sure the short versions are a good idea. It's tempting but I think that the slight gains in efficiency are not worth the misrecognitions and increase in complexity.

If users become familiar with the system and then want to tailor it to their own workflow with commands like these then that's great, but I think it's better to leave that decision up to them rather than making the system more complex for new users.

**Comment by kendonB** [comment]
Does "backslash" actually work for anyone in CCR? I find that dragon always recognises it as dictation and not as a caster command. I propose changing to "lean slash". "backslash" will still work just not with CCR.

**Comment by kendonB** [comment]
I think this is ready. Can someone else test? @alexboche?

**Comment by kendonB** [comment]
I'm finding "bakes" often gets recognized when I mean "backslash". I will change this to "ticky pair" unless anyone has any bright ideas for 2 syllable alternatives.

File: castervoice/lib/ccr/core/punctuation.py [hunk 1]
```diff
@@ -7,101 +7,92 @@
 from castervoice.lib.dfplus.merge.mergerule import MergeRule
 from castervoice.lib.dfplus.state.short import R
 
+text_punc_dict = {
+    "ace":                                  " ",
+    "clamor":                               "!",     
+    "chocky":                              "\"",
```
[the remark this answers is not in the document]
Comment (kendonB) [reply castervoice/lib/ccr/core/punctuation.py]: No it's right. chocky is the single double quote character: " or '"'

**Pull request merged**